src/rule_definition_tools.jl

```diff
@@ -80,16 +80,16 @@ macro scalar_rule(call, maybe_setup, partials...)
         end
     end
     if all(Meta.isexpr(partial, :tuple) for partial in partials)
-        forward_rules = Any[rule_from_partials(partial.args...) for partial in partials]
+        forward_rules = Any[rule_from_partials(input, partial.args...) for (input, partial) in zip(inputs, partials)]
         reverse_rules = Any[]
         for i in 1:length(inputs)
             reverse_partials = [partial.args[i] for partial in partials]
-            push!(reverse_rules, rule_from_partials(reverse_partials...))
+            push!(reverse_rules, rule_from_partials(inputs[i], reverse_partials...))
         end
     else
         @assert length(inputs) == 1 && all(!Meta.isexpr(partial, :tuple) for partial in partials)
-        forward_rules = Any[rule_from_partials(partial) for partial in partials]
-        reverse_rules = Any[rule_from_partials(partials...)]
+        forward_rules = Any[rule_from_partials(input, partial) for (input, partial) in zip(inputs, partials)]
```

Comment:
This leads to incorrect behavour.
because `length(inputs)` != `length(partials)`
Partials has same length as the outputs, not the inputs.

So for functions with fewer inputs than outputs it crops the number of rules to be only the number of inputs.

`sincos` is example of something this screws up.
I have not investigated the correct way to do this,